test/integration/full/error-occurred/error-ruleset.js

```diff
@@ -0,0 +1,55 @@
+window.assertIsErrorOccurred = function (result, { message, target }) {
+  assert.isDefined(result);
+  assert.isDefined(result.error);
+  assert.include(result.error.message, message);
+
+  assert.lengthOf(result.nodes, 1);
+  const node = result.nodes[0];
+  assert.lengthOf(node.any, 0);
+  assert.lengthOf(node.all, 0);
+  assert.lengthOf(node.none, 1);
+  assert.equal(node.none[0].id, 'error-occurred');
+  assert.include(node.none[0].message, 'Axe encountered an error');
+  assert.deepEqual(node.target, target);
```

Comment:
This is missing a few key checks that are currently failing:

```suggestion
  assert.isDefined(result);

  assert.isDefined(result.error);
  assert.include(result.error.message, message);
  assert.isDefined(result.error.method);
  assert.isUndefined(result.error.errorNode); // would be redundant with result.errorNode

  assert.isDefined(result.errorNode);
  assert.isUndefined(result.errorNode._element);
  assert.deepEqual(result.errorNode.target, target);
  assert.isDefined(result.errorNode.html);

  assert.lengthOf(result.nodes, 1);
  const node = result.nodes[0];
  assert.lengthOf(node.any, 0);
  assert.lengthOf(node.all, 0);
  assert.lengthOf(node.none, 1);
  assert.equal(node.none[0].id, 'error-occurred');
  assert.include(node.none[0].message, 'Axe encountered an error');
  assert.deepEqual(node.none[0].data, result.error);
  assert.deepEqual(node.target, target);
  assert.isDefined(node.html);
```